packages/react/src/ui/Connect/components/Connector/Connecting.tsx

```diff
@@ -27,17 +27,13 @@ export function Connecting({ className }: ConnectorProps) {
     isConnected,
   } = useConnectUI();
 
-  if (!connector) return null;
+  useEffect(() => {
+    if (isConnected && route === Routes.CONNECTING && !isConnecting) {
+      cancel();
```

Comment:
If we properly call cancel on all disconnections shouldn't we no longer need this?